ts/Stock/StockTools/StockToolsGui.ts

```diff
@@ -144,20 +143,29 @@ function onChartAfterGetContainer(
 function onChartBeforeRedraw(
     this: Chart
 ): void {
-    if (this.stockTools) {
+    onChartRedraw.call(this);
+    if (this.stockTools && this.stockTools.guiEnabled) {
```

Comment:
Can we change it to optional chaining?
```suggestion
    if (this.stockTools?.guiEnabled) {
```